src/test/tv/codely/scala_intro_examples/lesson_09_oop/exercise_solutions/joan_miralles_solution/EmailTest.scala

```diff
@@ -0,0 +1,46 @@
+package tv.codely.scala_intro_examples.lesson_09_oop.exercise_solutions.joan_miralles_solution
+
+import java.security.InvalidParameterException
+
+import org.scalatest.{Matchers, WordSpec}
+
+class EmailTest extends WordSpec with Matchers {
+
+  private val validEmail = "some@email.com"
+  private val anotherValidEmail = "a@e.t"
+  private val invalidEmail = "some.email.com"
+  private val anotherInvalidEmail = "some@email"
+
+  "Email Class" should {
+    "provide an apply method in the companion object in order to construct new Email from complete string" in {
+      "Email(completeEmail = validEmail)" should compile
```

Comment:
Instead of using the `should compile` matcher which make us to define code as strings which wouldn't be changed with automatic refactoring, nor checked at `test:compile` time, we could just assume that this code doesn't throw any exception.

That is, form:
```scala
"Email(completeEmail = validEmail)" should compile
```

To:
```scala
Email(completeEmail = validEmail)
```

This way, if that code desn't compile will fail at `test:compile` time, or if it throws an exception in runtime, will make the test fail 🙂